### PR TITLE
Support rc version in update_version.py

### PR DESCRIPTION
**Pull request by haon4**

### CODE REVIEW

File: update_version.py [hunk 1]
```diff
@@ -1,16 +1,25 @@
 #!/usr/bin/env python
+# Usage: ./update_version.py <MAJOR>.<MINOR>.<MICRO> [<RC version>]
+#
+# Example:
+# ./update_version.py 3.7.1 2
+#   => Version will become 3.7.1-rc-2 (beta)
+# ./update_version.py 3.8.1
+#   => Version will become 3.7.1-rc-2 (stable)
```
Comment (TeBoring) [inline update_version.py]: 3.8.1?

File: update_version.py [hunk 2]
```diff
@@ -120,44 +140,44 @@ def UpdateCsharp():
   RewriteXml('csharp/src/Google.Protobuf/Google.Protobuf.csproj',
     lambda document : ReplaceText(
       Find(Find(document.documentElement, 'PropertyGroup'), 'VersionPrefix'),
-      NEW_VERSION),
+      GetFullVersion()),
```
Comment (TeBoring) [inline update_version.py]: -rc.

File: update_version.py [hunk 3]
```diff
@@ -120,44 +140,44 @@ def UpdateCsharp():
...
     add_xml_prefix=False)
 
   RewriteXml('csharp/Google.Protobuf.Tools.nuspec',
     lambda document : ReplaceText(
       Find(Find(document.documentElement, 'metadata'), 'version'),
-      NEW_VERSION))
+      GetFullVersion()))
```
Comment (TeBoring) [inline update_version.py]: -rc.

File: update_version.py (hunk 1 above)
Comment (haon4) [reply update_version.py]: Done.

File: update_version.py (hunk 2 above)
Comment (haon4) [reply update_version.py]: Done.

File: update_version.py (hunk 3 above)
Comment (haon4) [reply update_version.py]: Done

**Comment by TeBoring** [comment]
Also added script for ```php/ext/google/protobuf/protobuf.h``` and ```java/bom/pom.xml``` in #5749 . They also need this change.

**Comment by haon4** [comment]
Update version for php/ext/google/protobuf/protobuf.h. java/bom/pom.xml is already being updated so no change there.

**Pull request merged**